src/rgw/rgw_sync.cc

```diff
@@ -1344,9 +1258,10 @@ class RGWMetaSyncShardCR : public RGWCoroutine {
             sync_marker.marker = sync_marker.next_step_marker;
             sync_marker.next_step_marker.clear();
           }
+          // XXX: why write the marker if !can_adjust_marker?
```

Comment:
@cbodley I think you have a point here, probably want to skip writing the marker in that case